### PR TITLE
Resolve pipeline downstreams

### PR DESCRIPTION
**Pull request by res0nance**
Signed-off-by: Raihaan Shouhell <raihaan.shouhell@autodesk.com>

This ensures that pipeline projects resolve instead of being unresolved

image

### CODE REVIEW

**Comment by res0nance** [comment]
Gentle ping @oleg-nenashev if you are cleaning up this plugin I'd like to have this included in the next release since this improves the user experience by allowing users to have a link to their downstream projects.

**Comment by oleg-nenashev** [comment]
Sorry, was not in my review queue

**Comment by res0nance** [comment]
@oleg-nenashev any plans to move this forward?

File: src/main/java/hudson/plugins/parameterizedtrigger/SubProjectData.java [hunk 1]
```diff
@@ -44,23 +44,23 @@
...
+    private final Set<Job> triggered = new TreeSet<>(customComparator);
     private final Set<String> unresolved = new TreeSet<String>();
 
     /**
      * A set intended to hold dynamically configured sub project
      *
      * @return  A set reserved for dynamically configured sub project
      */
-    public Set<AbstractProject> getDynamic() {
+    public Set<Job> getDynamic() {
```
Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/parameterizedtrigger/SubProjectData.java:63]: It breaks binary compatibility. Would be better to introduce new methods

Comment (olamy) [reply src/main/java/hudson/plugins/parameterizedtrigger/SubProjectData.java:63]: Looks fine as it's probably only use internally.

**Comment by olamy** [comment]
@res0nance some conflicts now. can you rebase please?

**Comment by res0nance** [comment]
@olamy I have rebased it

**Comment by merlindam** [comment]
If I'm correct this fix the issue [JENKINS-48714](https://issues.jenkins.io/browse/JENKINS-48714)
I'm not sure who should close the issue ? cc @res0nance @olamy

**Comment by darxriggs** [comment]
Thanks for this change. Now downstream pipelines are shown when they are triggered in a "build step".

I was wondering about another case: Triggering in a "post-build action". Any thoughts on this?

**Comment by res0nance** [comment]
> Thanks for this change. Now downstream pipelines are shown when they are triggered in a "build step".
> 
> I was wondering about another case: Triggering in a "post-build action". Any thoughts on this?

Hey Rene, not sure what you mean by this. Perhaps a screenshot would help

**Comment by darxriggs** [comment]
There are two options to use this plugin to trigger other jobs: "build step" and "post-build action"
I have attached three screenshots for each option: "configuration", "project view" and "build view"
Please have a look at the inconsistencies of "project view" and "build view".

# Observations

## Subprojects

build step - build view
* The "pipeline-triggered" link is missing.
* The "freestyle-triggered" link is functional but not containing the folder (as on the "project view").

post-build action - project view
* The "pipeline-triggered" link is missing.
* The "freestyle-triggered" link is missing.

post-build action - build view
* The "pipeline-triggered" link is missing.
* The "freestyle-triggered" link is missing.

## Downstream projects

build step - project view
* The "pipeline-triggered" link is missing.

post-build action - project view
* The "pipeline-triggered" link is missing.

# Option 1) build step

## configuration
screenshot freestyle-trigger-build-step config

## project view
screenshot freestyle-trigger-build-step project

## build view
screenshot freestyle-trigger-build-step build

# Option 2) post-build action

## configuration
screenshot freestyle-trigger-post-build-action config

## project view
screenshot freestyle-trigger-post-build-action project

## build view
screenshot freestyle-trigger-post-build-action build

**Pull request merged**